openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -0,0 +1,259 @@
+        # load will be a cache hit, so the course will be loaded from the cache.
+        course_overview_cache_miss = CourseOverview.get_from_id(course.id)
+        course_overview_cache_hit = CourseOverview.get_from_id(course.id)
+
+        # Test if value of these attributes match between the three objects
+        fields_to_test = [
+            'id',
+            'display_name',
+            'display_number_with_default',
+            'display_org_with_default',
+            'advertised_start',
+            'facebook_url',
+            'social_sharing_url',
+            'certificates_display_behavior',
+            'certificates_show_before_end',
+            'cert_name_short',
+            'cert_name_long',
+            'lowest_passing_grade',
+            'end_of_course_survey_url',
+            'mobile_available',
+            'visible_to_staff_only',
+            'location',
+            'number',
+            'url_name',
+            'display_name_with_default',
+            'start_date_is_still_default',
+            'pre_requisite_courses',
+        ]
+        for attribute_name in fields_to_test:
+            course_value = getattr(course, attribute_name)
+            cache_miss_value = getattr(course_overview_cache_miss, attribute_name)
+            cache_hit_value = getattr(course_overview_cache_hit, attribute_name)
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Test if return values for all methods are equal between the three objects
+        methods_to_test = [
+            ('clean_id', ()),
+            ('clean_id', ('#',)),
+            ('has_ended', ()),
+            ('has_started', ()),
+            ('start_datetime_text', ('SHORT_DATE',)),
+            ('start_datetime_text', ('DATE_TIME',)),
+            ('end_datetime_text', ('SHORT_DATE',)),
+            ('end_datetime_text', ('DATE_TIME',)),
+            ('may_certify', ()),
+        ]
+        for method_name, method_args in methods_to_test:
+            course_value = getattr(course, method_name)(*method_args)
+            cache_miss_value = getattr(course_overview_cache_miss, method_name)(*method_args)
+            cache_hit_value = getattr(course_overview_cache_hit, method_name)(*method_args)
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Other values to test
+        # Note: we test the start and end attributes here instead of in
+        # fields_to_test, because I ran into trouble while testing datetimes
+        # for equality. For Mongo Draft courses, start and end times ended up
+        # being fractions of a second different between course_value and
+        # cache_miss_value (implying a problem with the modulestore, not with
+        # this app). So, as a workaround, we simply test if the start and end
```

Comment:
Could you please remove this implication? The modulestore has 99+ problems but this ain't one.